lib/WebSocket.js

```diff
@@ -214,7 +228,7 @@ class WebSocket extends EventEmitter {
     this.extensions = null;
 
     this.removeAllListeners();
-    this.on('error', function onerror () {}); // catch all errors after this
+    this.on('error', noop); // Catch all errors after this.
```

Comment:
I _think_ the only cases where an error event may be emitted here are when trying to write to a socket that has already been closed due to write operation being queued in the `Sender`. In particular.

1. https://github.com/websockets/ws/blob/549cbe11527320f123e423bb13a1b20bc1807874/lib/WebSocket.js#L266.

2. https://github.com/websockets/ws/blob/549cbe11527320f123e423bb13a1b20bc1807874/lib/WebSocket.js#L159.

Maybe we should remove all queued operations in `Sender` when `finalize` is called.